File: problems/1207/solution.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        d, s = defaultdict(int), set()
        
        for num in arr:
            d[num] += 1
        
        for v in d.values():
            if v in s:
                return False
            
            s.add(v)
        
        return True

File: problems/1207/solution.go
package main
func uniqueOccurrences(arr []int) bool {
    m, count := map[int]int{}, map[int]struct{}{}
    
    for _, num := range arr {
        m[num]++
    }
    
    for _, v := range m {
        if _, ok := count[v]; ok {
            return false
        }
        
        count[v] = struct{}{}
    }
    
    return true
}